src/gui/search/searchsortmodel.cpp

```diff
@@ -29,17 +29,89 @@
 #include "searchsortmodel.h"
 
 SearchSortModel::SearchSortModel(QObject *parent)
-    : QSortFilterProxyModel(parent)
+    : base(parent)
+    , m_isNameFilterEnabled(false)
+    , m_minSeeds(0)
+    , m_maxSeeds(-1)
+    , m_minLeechs(0)
+    , m_maxLeechs(-1)
+    , m_minSize(0)
+    , m_maxSize(-1)
 {
 }
 
+void SearchSortModel::enableNameFilter(bool enable)
+{
+    m_isNameFilterEnabled = enable;
+}
+
+void SearchSortModel::setNameFilter(const QString &searchTerm)
+{
+    m_searchTerm = searchTerm;
+    if (searchTerm.length() > 2 &&
+            searchTerm.at(0) == QChar('\"') && searchTerm.at(searchTerm.length() - 1) == QChar('\"')) {
+        m_searchTermWords.append(m_searchTerm.mid(1, m_searchTerm.length() - 2));
+    }
+    else {
+        m_searchTermWords = searchTerm.split(QChar(' '), QString::SkipEmptyParts);
+    }
+}
+
+void SearchSortModel::setSizeFilter(qint64 minSize, qint64 maxSize)
+{
+    m_minSize = std::max(static_cast<qint64>(0), minSize);
```

Comment:
To my sadness, I still can not understand what do you want to tell me.
My point is the following. I know there has to be a function `max()` which accepts two objects of the same type. I don't know whether the best match is provided by a function template or there is a regular  overloaded function. To allow a compiler to include regular functions into the resolution list, I do not use template qualification, instead I cast the parameter.